File: backend/internal/models/lecture.go
package models

import (
	"time"
)

type Lecture struct {
	ID        int     `gorm:"primaryKey;autoIncrement"`
	Group     *string `gorm:"type:text"`
	Lector    *string `gorm:"type:text"`
	Platform  *string `gorm:"type:text"`
	Unit      *string `gorm:"type:text"`
	Location  *string `gorm:"type:text"`
	URL       *string `gorm:"type:text"`
	ShortURL  *string `gorm:"type:text"`
	StreamKey *string `gorm:"type:text"`

	Description *string `gorm:"type:text"`
	Admin       *string `gorm:"type:text;"`

	Date         time.Time `gorm:"not null"`
	Start        *string   `gorm:"type:text"`
	End          *string   `gorm:"type:text"`
	AbnormalTime *string   `gorm:"type:text"`

	CreatedAt time.Time  `gorm:"autoCreateTime"`
	UpdatedAt *time.Time `gorm:"autoUpdateTime"`
}

File: frontend/src/interfaces/response/lecture.ts
export interface LectureResponse {
  id: number;
  date: string;

  group?: string;
  lector?: string;
  platform?: string;
  unit?: string;
  location?: string;
  url?: string;
  shortUrl?: string;
  streamKey?: string;
  description?: string;
  admin?: string;
  start?: string;
  end?: string;
  abnormalTime?: string;
  createdAt?: string;
  updatedAt?: string;
}

export type LecturesResponse = LectureResponse[];
